### PR TITLE
Teaches Auto-GPT to use DB connection and SQL execution

### PR DESCRIPTION
**Pull request by KaneCohen**
Adds new flag `--db` which tells Auto-GPT to try and establish connection with common mysql-like databases and then perform couple of basic functions:

- Reading information about specific table
- Based on table data create and execute SQL created a given user request

Beware of little Bobby Tables.

### CODE REVIEW

**Review by ZsoltGergely — APPROVED** [review]

File: scripts/commands.py [hunk 1]
```diff
@@ -19,15 +20,15 @@
 def get_command(response):
     try:
         response_json = fix_and_parse_json(response)
-
```
Comment (nponeccop) [inline scripts/commands.py]: Submit the whitespace fixes in a separate PR

File: scripts/json_parser.py [hunk 2]
```diff
@@ -73,4 +73,4 @@ def fix_json(json_str: str, schema: str, debug=False) -> str:
         # import traceback
         # call_stack = traceback.format_exc()
         # print(f"Failed to fix JSON: '{json_str}' "+call_stack)
-        return "failed"
```
Comment (nponeccop) [inline scripts/json_parser.py]: Note that this is also a whitespace fix

**Review by nponeccop — CHANGES_REQUESTED** [review]

**Comment by nponeccop** [comment]
@KaneCohen Resolve the conflicts and make the PR more focused

File: scripts/commands.py [hunk 3]
```diff
@@ -58,6 +58,7 @@ def execute_command(command_name, arguments):
 
     try:
         if command_name == "google":
+
```
Comment (nponeccop) [inline scripts/commands.py]: Unrelated whitespace change

File: scripts/config.py [hunk 4]
```diff
@@ -98,5 +116,15 @@ def set_google_api_key(self, value: str):
     def set_custom_search_engine_id(self, value: str):
         self.custom_search_engine_id = value
 
+
```
Comment (nponeccop) [inline scripts/config.py:134]: More whitespace

File: scripts/main.py [hunk 5]
```diff
@@ -273,6 +273,11 @@ def parse_arguments():
         init_db()
         print_to_console("DB Mode: ", Fore.GREEN, "ENABLED")
 
+    if args.debug:
+        print_to_console("Debug Mode: ", Fore.GREEN, "ENABLED")
+        cfg.set_debug_mode(True)
+
+
```
Comment (nponeccop) [inline scripts/main.py:280]: Too many blank lines

**Review by nponeccop — CHANGES_REQUESTED** [review]

**Comment by nponeccop** [comment]
There are conflicts again :-\

**Comment by nponeccop** [comment]
Well, I don't know how to review it, the "condensed view" confuses me. @ryanpeach ?

**Comment by nponeccop** [comment]
@KaneCohen Can you resolve the conflicts?

File: .env.template [hunk 6]
```diff
@@ -12,4 +12,10 @@ OPENAI_AZURE_API_VERSION=api-version-for-azure
 OPENAI_AZURE_DEPLOYMENT_ID=deployment-id-for-azure
 IMAGE_PROVIDER=dalle
 HUGGINGFACE_API_TOKEN=
+DB_HOST=
+DB_PORT=
+DB_USERNAME=
+DB_PASSWORD=
+DB_DATABASE=
 USE_MAC_OS_TTS=False
+
```
Comment (nponeccop) [inline .env.template:21]: Remove the extra empty line

File: scripts/config.py [hunk 7]
```diff
@@ -78,6 +87,7 @@ def __init__(self):
         # Note that indexes must be created on db 0 in redis, this is not configureable.
 
         self.memory_backend = os.getenv("MEMORY_BACKEND", 'local')
+
```
Comment (nponeccop) [inline scripts/config.py:90]: Here too. Don't reformat the code you don't change

File: scripts/main.py [hunk 8]
```diff
@@ -292,6 +295,15 @@ def parse_arguments():
...
 
+    if args.db:
+        cfg.set_db_access(True)
+        init_db()
+        print_to_console("DB Mode: ", Fore.GREEN, "ENABLED")
+
+    if args.debug:
+        print_to_console("Debug Mode: ", Fore.GREEN, "ENABLED")
+        cfg.set_debug_mode(True)
+
```
Comment (nponeccop) [inline scripts/main.py:306]: Two empty lines is enough. Remove the third one.

**Review by nponeccop — APPROVED** [review]
Good job. Now a few whitespace fixes.

**Comment by Wladastic** [comment]
Was this abandoned?

**Comment by KaneCohen** [comment]
> Was this abandoned?

Hey, given the rate of change on a main repo - seemed like a good idea to let it stabilise first.

**Comment by ntindle** [comment]
This would be a great plug-in!

**Comment by ntindle** [comment]
Hello!  We sincerely appreciate your contribution to our project. We encourage you to open a pull request in the Auto-GPT-Plugins repository.  If you encounter any difficulties in converting your code into a plugin, please do not hesitate to seek assistance from the #plugins-chat on our Discord Server.  We look forward to your contribution to the Plugins repository and apologize for any inconvenience this may have caused.

**Pull request closed without merging**